packages/@ourworldindata/grapher/src/core/Grapher.tsx

```diff
@@ -1198,6 +1199,9 @@ export class Grapher
     }
 
     @computed get availableTabs(): GrapherTabOption[] {
+        const { shownTabs } = this
+        if (shownTabs && shownTabs.length > 0) return shownTabs
```

Comment:
I'm not entirely sure about this; maybe it should rather be the intersection between `shownTabs` and the below code?
I.e., if `shownTabs` contains `chart`, but `hasChartTab` is false, then we wouldn't show the chart tab anyhow?

Reply:
Yeah this is a bit tricky. I thought that maybe having this as a definitive set if it is given is clearer. Alternatively I was thinking of retiring the hasMapTab and hasChartTab and replace them by a hiddenTabs enumeration. The reason why I didn't want to remove hasMapTab and hasChartTab without something like hiddenTabs is that it would mean that the migration to remove these would have to materialize the current tab options into every chart config and that doesn't seem so nice.

The logic with hiddenTabs would be something like:
```javascript
const initialTabs = config.shownTabs ?? ALL_GRAPHER_TABS
const availablTabs = _.difference(initialTabs, config.hiddenTabs)
```
This would give us the full flexibility and a clear way to migrate this setting without having to commit all current tabs into the grapher configs.

What do you think?

Reply:
Yeah that makes sense.
However, we use `hasChartTab` and `hasMapTab` in quite some places today, so we probably want to keep it as a computed property.